crates/bevy_render/src/rangefinder.rs

```diff
@@ -0,0 +1,53 @@
+use bevy_math::{Mat4, Vec4};
+
+use crate::view::ExtractedView;
+
+/// A distance calculator for the draw order of [`PhaseItem`](crate::render_phase::PhaseItem)s.
+pub struct ViewRangefinder3d {
```

Comment:
Is there a reason this has `3d` in the name?
It's useless for 2d since you don't have to worry about the view rotation, but it would technically work, I think?
If it does work fine for 2d (even if we don't use it there) I'd prefer to remove 3d from the name.

Reply:
The rationale for the name is: A PhaseItem impl can use any metric it likes for its sort key. The ViewRangefinder3d calculates values which are intended for use with Opaque3d, AlphaMask3d, and Transparent3d.

Code which enqueues Transparent2d items seems to assume that the projection matrix doesn't do anything interesting to the Z coordinate and just uses the Z translation from the model transform matrix as the sort key. Therefore, I think, the distance values generated by ViewRangefinder3d would not be consistent with the rest of the 2D pipeline.

Reply:
Thinking about it again it is a good idea to associate this with 3d stuff as 2d a 3d may grow apart more in the future anyways